spatialos-sdk/src/worker/component.rs

```diff
@@ -115,28 +118,35 @@ impl<'a> ComponentDataRef<'a> {
     }
 }
 
-// #[derive(Debug)]
-// pub struct ComponentUpdate<'a> {
-//     pub component_id: ComponentId,
-//     pub schema_type: SchemaComponentUpdate<'a>,
-//     pub user_handle: *const Worker_ComponentUpdateHandle,
+#[derive(Debug)]
+pub(crate) struct ComponentUpdateRef<'a> {
+    pub component_id: ComponentId,
+    pub schema_type: &'a schema::ComponentUpdate,
+    pub user_handle: *mut Worker_ComponentUpdateHandle,
+}
 
-//     // NOTE: `user_handle` is borrowing data owned by the parent object, but it's a
-//     // type-erased pointer that may be null, so we just mark that we're borrowing
-//     // *something*.
-//     pub _marker: PhantomData<&'a ()>,
-// }
+impl<'a> ComponentUpdateRef<'a> {
+    pub unsafe fn from_raw(data: &'a Worker_ComponentUpdate) -> Self {
```

Comment:
I had a couple of reasons for moving away from the `From` impl:

* The main reason is that the conversion function needs to be marked as unsafe. Since the wrapper type is meant to be a safe abstraction over the C API, its safety depends on it being correctly constructed from valid data.
* Trait impls are public, whereas it's important that the conversions be private to the crate in order to enforce safety (by ensuring that users can never construct an invalid object).
* `from_raw` more accurately reflects the nature of the function, that it's a conversion from the raw C representation into a safe Rust representation. This is a more minor point, but it feels like `From` isn't the correct idiom in this case, so an explicit trait method seems to me like the better approach.